--- configure
+++ configure
@@ -16772,6 +16772,7 @@
 rm -f core conftest.err conftest.$ac_objext conftest.beam \
     conftest$ac_exeext conftest.$ac_ext
 
+if test "$enable_pcregrep_libz" = "yes"; then
 # Check for the availability of libz (aka zlib)
 
        for ac_header in zlib.h
@@ -16823,8 +16824,10 @@
 then :
   HAVE_LIBZ=1
 fi
+fi
 
 
+if test "$enable_pcregrep_libbz2" = "yes"; then
 # Check for the availability of libbz2. Originally we just used AC_CHECK_LIB,
 # as for libz. However, this had the following problem, diagnosed and fixed by
 # a user:
@@ -16885,6 +16888,7 @@
 rm -f core conftest.err conftest.$ac_objext conftest.beam \
     conftest$ac_exeext conftest.$ac_ext
 LIBS="$OLD_LIBS"
+fi
 
 # Check for the availability of libreadline
 
