--- pcre2-config.in
+++ pcre2-config.in
@@ -3,6 +3,8 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 exec_prefix_set=no
+includedir=@includedir@
+libdir=@libdir@
 
 cflags="[--cflags]"
 libs=
@@ -30,15 +32,15 @@
 libR=
 case `uname -s` in
   *SunOS*)
-  libR=" -R@libdir@"
+  libR=" -R${libdir}"
   ;;
   *BSD*)
-  libR=" -Wl,-R@libdir@"
+  libR=" -Wl,-R${libdir}"
   ;;
 esac
 
 libS=
-case "@libdir@" in
+case "${libdir}" in
   # system locations on Linux, no -L needed
   /usr/lib|/usr/lib32|/usr/lib64|/usr/lib/*-gnu*)
   ;;
@@ -48,7 +50,7 @@
   /usr/lib/64|/usr/lib/amd*|/usr/lib/sparc*|/usr/lib/ppc*)
   ;;
   *)
-  libS="-L@libdir@"
+  libS="-L${libdir}"
   ;;
 esac
 
@@ -79,15 +81,15 @@
       echo @PACKAGE_VERSION@
       ;;
     --cflags)
-      if test @includedir@ != /usr/include ; then
-        includes=-I@includedir@
+      if test ${includedir} != /usr/include ; then
+        includes=-I${includedir}
       fi
       echo $includes @PCRE2_STATIC_CFLAG@
       ;;
     --cflags-posix)
       if test @enable_pcre2_8@ = yes ; then
-        if test @includedir@ != /usr/include ; then
-          includes=-I@includedir@
+        if test ${includedir} != /usr/include ; then
+          includes=-I${includedir}
         fi
         echo $includes @PCRE2POSIX_CFLAG@
       else
