--- bundled/linux/include/uapi/linux/taskstats.h
+++ bundled/linux/include/uapi/linux/taskstats.h
@@ -18,7 +18,6 @@
 #define _LINUX_TASKSTATS_H
 
 #include <linux/types.h>
-#include <linux/time_types.h>
 
 /* Format for per-task data returned to userland when
  *	- a task exits
@@ -35,7 +34,7 @@
  */
 
 
-#define TASKSTATS_VERSION	17
+#define TASKSTATS_VERSION	16
 #define TS_COMM_LEN		32	/* should be >= TASK_COMM_LEN
 					 * in linux/sched.h */
 
@@ -232,15 +231,6 @@
 	__u64	irq_delay_max;
 	__u64	irq_delay_min;
 
-	/*v17: delay max timestamp record*/
-	struct __kernel_timespec cpu_delay_max_ts;
-	struct __kernel_timespec blkio_delay_max_ts;
-	struct __kernel_timespec swapin_delay_max_ts;
-	struct __kernel_timespec freepages_delay_max_ts;
-	struct __kernel_timespec thrashing_delay_max_ts;
-	struct __kernel_timespec compact_delay_max_ts;
-	struct __kernel_timespec wpcopy_delay_max_ts;
-	struct __kernel_timespec irq_delay_max_ts;
 };
 
 
