--- src/net.c
+++ src/net.c
@@ -20,7 +20,7 @@
 
 #ifndef WIN32
 #  include <arpa/inet.h>
-#  ifndef _AIX
+#  if !defined(_AIX) && defined(HAVE_IFADDRS_H)
 #    include <ifaddrs.h>
 #  endif
 #  include <netdb.h>
@@ -718,7 +718,7 @@
 }
 
 
-#if !defined(WIN32) && !defined(_AIX)
+#if !defined(WIN32) && !defined(_AIX) && defined(HAVE_IFADDRS_H)
 
 
 static int net__bind_interface(struct mosquitto__listener *listener, struct addrinfo *rp)
@@ -874,7 +874,7 @@
 			return 1;
 		}
 
-#if !defined(WIN32) && !defined(_AIX)
+#if !defined(WIN32) && !defined(_AIX) && defined(HAVE_IFADDRS_H)
 		if(listener->bind_interface){
 			/* It might be possible that an interface does not support all relevant sa_families.
 			 * We should successfully find at least one. */
@@ -917,7 +917,7 @@
 	}
 	freeaddrinfo(ainfo);
 
-#ifndef WIN32
+#if !defined(WIN32) && defined(HAVE_IFADDRS_H)
 	if(listener->bind_interface && !interface_bound){
 		mosquitto_FREE(listener->socks);
 		return 1;
