--- Makefile.am
+++ Makefile.am
@@ -95,8 +95,8 @@
 
 TAGS_DEPENDENCIES = $(CURL_ETAGS_FILES)
 
-SUBDIRS = lib docs src scripts
-DIST_SUBDIRS = $(SUBDIRS) tests projects include docs
+SUBDIRS = lib      src scripts
+DIST_SUBDIRS = $(SUBDIRS)       projects include
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
@@ -174,14 +174,11 @@
 else
 install-data-hook:
 	(cd include && $(MAKE) install)
-	(cd docs && $(MAKE) install)
 endif
 
 # We extend the standard uninstall with a custom hook:
 uninstall-hook:
 	(cd include && $(MAKE) uninstall)
-	(cd docs && $(MAKE) uninstall)
-	(cd docs/libcurl && $(MAKE) uninstall)
 
 ca-bundle: $(srcdir)/scripts/mk-ca-bundle.pl
 	@echo "generating a fresh ca-bundle.crt"
--- Makefile.in
+++ Makefile.in
@@ -513,8 +513,8 @@
 
 ETAGS_ARGS = $(CURL_ETAGS_FILES)
 TAGS_DEPENDENCIES = $(CURL_ETAGS_FILES)
-SUBDIRS = lib docs src scripts
-DIST_SUBDIRS = $(SUBDIRS) tests projects include docs
+SUBDIRS = lib      src scripts
+DIST_SUBDIRS = $(SUBDIRS)       projects include
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
 all: all-recursive
@@ -1130,13 +1130,10 @@
 @BUILD_DOCS_TRUE@	(cd docs/libcurl && $(MAKE) install)
 @BUILD_DOCS_FALSE@install-data-hook:
 @BUILD_DOCS_FALSE@	(cd include && $(MAKE) install)
-@BUILD_DOCS_FALSE@	(cd docs && $(MAKE) install)
 
 # We extend the standard uninstall with a custom hook:
 uninstall-hook:
 	(cd include && $(MAKE) uninstall)
-	(cd docs && $(MAKE) uninstall)
-	(cd docs/libcurl && $(MAKE) uninstall)
 
 ca-bundle: $(srcdir)/scripts/mk-ca-bundle.pl
 	@echo "generating a fresh ca-bundle.crt"
