--- libbridge/Makefile.in
+++ libbridge/Makefile.in
@@ -1,11 +1,13 @@
 
-KERNEL_HEADERS=-I@KERNEL_HEADERS@
+KERNEL_HEADERS=@KERNEL_HEADERS@
 
 AR=ar
 RANLIB=@RANLIB@
 
 CC=@CC@
-CFLAGS = -Wall -g $(KERNEL_HEADERS)
+CFLAGS= @CFLAGS@
+CFLAGS += -I$(KERNEL_HEADERS)
+CFLAGS += $(COPTS)
 
 prefix=@prefix@
 exec_prefix=@exec_prefix@
@@ -34,8 +36,7 @@
 	$(RANLIB) $@
 
 %.o: %.c libbridge.h libbridge_private.h
-	$(CC) $(CFLAGS) $(INCLUDE) -c $<
+	$(CC) $(CFLAGS) -c $<
 
 libbridge_compat.o:	libbridge_compat.c if_index.c
 	$(CC) $(CFLAGS) -c libbridge_compat.c
-
--- brctl/Makefile.in
+++ brctl/Makefile.in
@@ -1,10 +1,11 @@
 
-KERNEL_HEADERS=-I@KERNEL_HEADERS@
+KERNEL_HEADERS=@KERNEL_HEADERS@
 
 CC=@CC@
-CFLAGS= -Wall @CFLAGS@
+CFLAGS= @CFLAGS@
+CFLAGS +=-I../libbridge -I$(KERNEL_HEADERS) 
+CFLAGS += $(COPTS)
 LDFLAGS=@LDFLAGS@
-INCLUDE=-I../libbridge $(KERNEL_HEADERS) 
 LIBS= -L ../libbridge -lbridge @LIBS@
 
 prefix=@prefix@
@@ -37,8 +38,7 @@
 	$(CC) $(LDFLAGS) $(brctl_OBJECTS) $(LIBS) -o brctl
 
 %.o: %.c brctl.h
-	$(CC) $(CFLAGS) $(INCLUDE) -c $< 
+	$(CC) $(CFLAGS) -c $<
 
 clean:
 	rm -f *.o brctl core
-
--- Makefile.in
+++ Makefile.in
@@ -14,10 +14,10 @@
 SUBDIRS=libbridge brctl doc
 
 all:
-	for x in $(SUBDIRS); do $(MAKE) $(MFLAGS) -C $$x ; done
+	for x in $(SUBDIRS); do $(MAKE) -C $$x ; done
 
 clean:
-	for x in $(SUBDIRS); do $(MAKE) $(MFLAGS) -C $$x clean ; done
+	for x in $(SUBDIRS); do $(MAKE) -C $$x clean ; done
 
 distclean:	clean
 	rm -f config.log
@@ -32,4 +32,3 @@
 
 install:
 	for x in $(SUBDIRS); do $(MAKE) $(MFLAGS) -C $$x install; done
-
